zipline/gens/brokers/alpaca_broker.py

```diff
@@ -0,0 +1,261 @@
+            limit=float(order.limit_price) if order.limit_price else None,
+            dt=order.submitted_at,
+            commission=0,
+        )
+        zp_order.status = ZP_ORDER_STATUS.OPEN
+        if order.canceled_at:
+            zp_order.status = ZP_ORDER_STATUS.CANCELLED
+        if order.failed_at:
+            zp_order.status = ZP_ORDER_STATUS.REJECTED
+        if order.filled_at:
+            zp_order.status = ZP_ORDER_STATUS.FILLED
+            zp_order.filled = int(order.filled_qty)
+        return zp_order
+
+    def _new_order_id(self):
+        return uuid.uuid4().hex
+
+    def order(self, asset, amount, style):
+        symbol = asset.symbol
+        qty = amount if amount > 0 else -amount
+        side = 'buy' if amount > 0 else 'sell'
+        order_type = 'market'
+        if isinstance(style, MarketOrder):
+            order_type = 'market'
+        elif isinstance(style, LimitOrder):
+            order_type = 'limit'
+        elif isinstance(style, StopOrder):
+            order_type = 'stop'
+        elif isinstance(style, StopLimitOrder):
+            order_type = 'stop_limit'
+
+        limit_price = style.get_limit_price(side == 'buy') or None
+        stop_price = style.get_stop_price(side == 'buy') or None
+
+        zp_order_id = self._new_order_id()
+        dt = pd.to_datetime('now', utc=True)
+        zp_order = ZPOrder(
+            dt=dt,
+            asset=asset,
+            amount=amount,
+            stop=stop_price,
+            limit=limit_price,
+            id=zp_order_id,
+        )
+
+        order = self._api.submit_order(
+            symbol=symbol,
+            qty=qty,
+            side=side,
+            type=order_type,
+            time_in_force='day',
+            limit_price=limit_price,
+            stop_price=stop_price,
+            client_order_id=zp_order.id,
+        )
+        zp_order = self._order2zp(order)
+        return zp_order
+
+    @property
+    def orders(self):
+        return {
+            o.client_order_id: self._order2zp(o)
+            for o in self._api.list_orders()
+        }
+
+    @property
+    def transactions(self):
+        return {
+            o.client_order_id: self._order2zp(o)
+            for o in self._api.list_orders(status='all')
+        }
+
+    def cancel_order(self, zp_order_id):
+        try:
+            order = self._api.get_order_by_client_order_id(zp_order_id)
+            self._api.cancel_order(order.id)
+        except Exception as e:
+            log.error(e)
+            return
+
+    def get_last_traded_dt(self, asset):
+        quote = self._api.get_quote(asset.symbol)
+        return pd.Timestamp(quote.last_timestamp)
+
+    def get_spot_value(self, assets, field, dt, data_frequency):
+        assert(field in (
+            'open', 'high', 'low', 'close', 'volume', 'price', 'last_traded'))
+        assets_is_scalar = not isinstance(assets, (list, set, tuple))
+        is_daily = 'd' in data_frequency
```

Comment:
I'd recommend to make an exact match to 'daily' here.

Reply:
I wasn't sure if it is "1d" or "daily". The zipline convention is sometimes confusing. This way it works either way ;) So it is "daily" here not "1d"?

Reply:
you are right: 1d is the proper representation. it's fine to keep it 'd', but a comment would be helpful to remove ambiguity.